### PR TITLE
python: Add new functions unique() and relabelConsecutive()

### PR DESCRIPTION
**Pull request by stuarteberg**
This PR adds two more functions that are helpful when dealing with huge label volumes.  This implementation of `unique()` is 3x faster than `numpy.unique()`. Similarly, this implementation of`relabelConsecutive()` is much faster than the best python implementation I could come up with.

### CODE REVIEW

File: vigranumpy/src/core/segmentation.cxx [hunk 1]
```diff
@@ -1147,6 +1148,65 @@ pythonApplyMapping(NumpyArray<NDIM, Singleband<SrcVoxelType> > src,
 //VIGRA_PYTHON_MULTITYPE_FUNCTOR_NDIM(pyApplyMapping, pythonApplyMapping)
 
 
+/** \brief Find unique values in the given array.
+*/
+template <class VoxelType, unsigned int NDIM>
+NumpyAnyArray
+pythonUnique(NumpyArray<NDIM, Singleband<VoxelType> > src)
+{
+    boost::unordered_set<VoxelType> labelset;
```
Comment (ukoethe) [inline vigranumpy/src/core/segmentation.cxx]: Is there a reason to prefer boost:: over std:: here?

File: vigranumpy/src/core/segmentation.cxx [hunk 2]
```diff
@@ -1147,6 +1148,65 @@ pythonApplyMapping(NumpyArray<NDIM, Singleband<SrcVoxelType> > src,
...
+                         NumpyArray<NDIM, Singleband<DestVoxelType> > res = NumpyArray<NDIM, Singleband<SrcVoxelType> >())
+{
+    using namespace boost::python;
+    res.reshapeIfEmpty(src.taggedShape(), "relabelConsecutive(): Output array has wrong shape.");
+
+    {
+        PyAllowThreads _pythread;
+
+        boost::unordered_map<SrcVoxelType, DestVoxelType> labelmap;
+        auto labelmap_end = labelmap.end();
```
Comment (ukoethe) [inline vigranumpy/src/core/segmentation.cxx]: IIRC, the end iterator can be invalidated by insert. Call .end() in the lambda.

File: vigranumpy/src/core/segmentation.cxx [hunk 3]
```diff
@@ -1147,6 +1148,65 @@ pythonApplyMapping(NumpyArray<NDIM, Singleband<SrcVoxelType> > src,
...
+
+VIGRA_PYTHON_MULTITYPE_FUNCTOR_NDIM(pyUnique, pythonUnique)
+
+
+/** \brief Relabel an array such that all labels are consecutive
+ * (i.e. there are no gaps in the label values used by the array)
+ * See python docstring below for details.
+*/
+template <unsigned int NDIM, class SrcVoxelType, class DestVoxelType>
+NumpyAnyArray
```
Comment (ukoethe) [inline vigranumpy/src/core/segmentation.cxx]: I think the function should also return the new max label and the mapping (converted to a Python dict).

**Comment by stuarteberg** [comment]
Thanks again for the careful review.  I've updated the code to address your suggestions.

**Pull request merged**